app/models/grader.rb

```diff
@@ -21,18 +23,29 @@ def self.unique
     select(column_names - ["id"]).distinct
   end
 
-  def grade(assignment, submission)
+  def really_grade(asn_id, sub_id)
+    puts "XX really grade"
+    puts self.inspect
+
+    assignment = Assignment.find(asn_id)
+    submission = Submission.find(sub_id)
+
     ans = do_grading(assignment, submission)
     submission.compute_grade! if submission.grade_complete?
-    ans
+  end
+
+  def grade(assignment, submission, prio = 0)
```

Comment:
And...the synchronous nature of it is ok to me, since I'm often trying to identify "is this thing hanging when I don't expect it to be hanging?"  So the web-request timeout, and the blocking of everyone else while I'm trying to diagnose a problem, is fine by me.